pkg/universal-cnf/config/removeClientInterface_test.go

```diff
@@ -0,0 +1,243 @@
+package config
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/cisco-app-networking/nsm-nse/pkg/universal-cnf/config/mocks"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connectioncontext"
+	"go.ligato.io/vpp-agent/v3/proto/ligato/vpp"
+	interfaces "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/interfaces"
+	l3 "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/l3"
+)
+
+func TestRemoveClientInterface(t *testing.T) {
+	cases := map[string]struct {
+		interfaces    []*interfaces.Interface
+		conn          *connection.Connection
+		routes        []*l3.Route
+		expectedError error
+	}{
+		"Remove from nil interface": {
+			nil,
+			conn1,
+			routes1,
+			fmt.Errorf("client interface with dstIpAddr %s not found", conn1.Context.IpContext.DstIpAddr),
+		},
+		"Remove from empty interface": {
+			[]*interfaces.Interface{},
+			conn1,
+			routes1,
+			fmt.Errorf("client interface with dstIpAddr %s not found", conn1.Context.IpContext.DstIpAddr),
+		},
+		"Remove from one interface": {
+			interfaces1,
+			conn1,
+			routes1,
+			nil,
+		},
+		"Remove from multiple interfaces": {
+			interfaces3,
+			conn1,
+			routes1,
+			nil,
+		},
+		"Unmatched NextHopAddr": {
+			interfaces1,
+			conn1,
+			routes2,
+			nil,
+		},
+		"Route prefix matches DstNetwork": {
+			interfaces1,
+			conn3,
+			routes1,
+			nil,
+		},
+		"Unmatched connection dstIpAddr": {
+			interfaces1,
+			conn2,
+			routes1,
+			fmt.Errorf("client interface with dstIpAddr %s not found", conn1.Context.IpContext.DstIpAddr),
+		},
+		"Unfound dstIpAddr in interfaces": {
+			interfaces2,
+			conn1,
+			routes1,
+			fmt.Errorf("client interface with dstIpAddr %s not found", conn1.Context.IpContext.DstIpAddr),
+		},
+	}
+
+	for testName, c := range cases {
+		t.Logf("Running test case: %s", testName)
+
+		ucnfBackend := &mocks.UniversalCNFBackend{}
+		uce := &UniversalCNFEndpoint{
+			backend:  ucnfBackend,
+			dpConfig: &vpp.ConfigData{Interfaces: c.interfaces, Routes: c.routes},
+		}
+
+		removeConfig, err := uce.removeClientInterface(c.conn)
+		if c.expectedError != nil {
+			assert.NotNil(t, err)
+		} else {
+			assert.Nil(t, err)
+			assert.NotNil(t, removeConfig)
+			assert.Equal(t, 1, len(removeConfig.Interfaces))
+		}
+	}
+}
```

Comment:
You can opt for a more elegant approach. https://play.golang.org/p/XnZoNIDliae. Your call.

Reply:
Thanks for the review! I will make changes accordingly